src/pages/AdminSettings/AdminMembers/MemberList/MemberTable.tsx

```diff
@@ -0,0 +1,237 @@
+  columnHelper.accessor('username', {
+    header: 'User Name',
+    cell: ({ renderValue }) => renderValue(),
+  }),
+  columnHelper.accessor('type', {
+    header: () => <span className="flex grow flex-row text-center">Type</span>,
+    cell: ({ renderValue }) => renderValue(),
+  }),
+  columnHelper.accessor('email', {
+    header: 'Email',
+    cell: ({ renderValue }) => renderValue(),
+  }),
+  columnHelper.accessor('activationStatus', {
+    header: 'Activation Status',
+    cell: ({ renderValue }) => renderValue(),
+  }),
+]
+
+interface MutationArgs {
+  activated: boolean
+  ownerid: number
+}
+
+interface CreateTableArgs {
+  tableData: UserListOwner[]
+  seatData: SelfHostedSettings
+  mutate: UseMutateFunction<any, unknown, MutationArgs, unknown>
+}
+
+const createTable = ({ tableData, seatData, mutate }: CreateTableArgs) => {
+  return tableData?.map(
+    ({ ownerid, activated, email, isAdmin, name, username }) => {
+      let maxSeats = seatData?.seatsUsed === seatData?.seatsLimit
+      const disableToggle = maxSeats && !activated
+
+      return {
+        username: <p>{name || username}</p>,
+        type: isAdmin ? 'Admin' : 'Developer',
+        email: <p>{email}</p>,
+        activationStatus: (
+          <Toggle
+            dataMarketing="handle-members-activation"
+            label={activated ? 'Activated' : 'Non-Active'}
+            value={activated}
+            onClick={() => {
+              mutate({ ownerid, activated: !activated })
+            }}
+            disabled={disableToggle}
+          />
+        ),
+      }
+    }
+  )
+}
+
+const Loader = () => (
+  <div className="mb-4 flex justify-center pt-4">
+    <Spinner />
+  </div>
+)
+
+function LoadMoreTrigger({
+  intersectionRef,
+}: {
+  intersectionRef: React.Ref<HTMLSpanElement>
+}) {
+  return (
+    <span
+      ref={intersectionRef}
+      className="invisible relative top-[-65px] block leading-[0]"
+    >
+      Loading
+    </span>
+  )
+}
+
+function MemberTable() {
+  const queryClient = useQueryClient()
+  const { ref, inView } = useInView()
+  const { data: seatData } = useSelfHostedSettings()
+  const { params } = useLocationParams({
+    activated: undefined,
+    isAdmin: undefined,
+    search: '',
+  })
+
+  const { data, isFetchingNextPage, hasNextPage, fetchNextPage, isLoading } =
+    useSelfHostedUserList(params)
+
+  const { mutate } = useMutation({
+    mutationFn: ({ activated, ownerid }: MutationArgs) =>
+      Api.patch({ path: `/users/${ownerid}`, body: { activated } }),
+    useErrorBoundary: true,
+    onSuccess: () => {
+      queryClient.invalidateQueries(['SelfHostedSettings'])
+      queryClient.invalidateQueries(['Seats'])
+      queryClient.invalidateQueries(['SelfHostedUserList'])
+    },
+  })
+
+  useEffect(() => {
+    if (inView && hasNextPage) {
+      fetchNextPage()
+    }
+  }, [fetchNextPage, inView, hasNextPage])
+
+  const tableContent = useMemo(() => {
+    if (!data || data.length === 0 || !seatData) {
```

Comment:
nit: I think the first 2 clauses can be combined into !data?.length, assuming that doesn't break a TS type